modules/blocks/chain.js

```diff
@@ -358,6 +358,13 @@ Chain.prototype.applyBlock = function (block, broadcast, saveBlock, cb) {
 		},
 		// Apply transactions to unconfirmed mem_accounts fields.
 		applyUnconfirmed: function (seriesCb) {
+
+			// If block is not to be saved, then we don't need to process unconfirmed state
+			// that block must be restored from the database for snapshot process
+			if(!saveBlock) {
```

Comment:
Please restore `applyUnconfirmed` for now, as we want snapshotting process to be as close as possible to sync process.